src/Support/Debug/Debug.php

```diff
@@ -0,0 +1,985 @@
+    {
+        $this->showBackTrace = $showBackTrace;
+
+        return $this;
+    }
+
+    /**
+     * Sets if files must be completely opened and showed in the output
+     * or just the fragment related to the exception
+     *
+     * @param bool $showFileFragment
+     *
+     * @return Debug
+     */
+    public function setShowFileFragment(bool $showFileFragment): Debug
+    {
+        $this->showFileFragment = $showFileFragment;
+
+        return $this;
+    }
+
+    /**
+     * Set if files part of the backtrace must be shown in the output
+     *
+     * @param bool $showFiles
+     *
+     * @return Debug
+     */
+    public function setShowFiles(bool $showFiles): Debug
+    {
+        $this->showFiles = $showFiles;
+
+        return $this;
+    }
+
+    /**
+     * Change the base URI for static resources
+     *
+     * @param string $uri
+     *
+     * @return $this
+     */
+    public function setUri(string $uri): Debug
+    {
+        $this->uri = $uri;
+
+        return $this;
+    }
+
+    /**
+     * Escapes a string with htmlentities
+     *
+     * @param mixed $value
+     *
+     * @return string
+     */
+    protected function escapeString($value): string
+    {
+        if (true === is_string($value)) {
+            return htmlentities(
+                str_replace("\n", "\\n", $value),
+                ENT_COMPAT,
+                'utf-8'
+            );
+        }
+
+        return $value;
+    }
+
+    /**
+     * Produces a recursive representation of an array
+     *
+     * @param array $arguments
+     * @param int   $number
+     *
+     * @return string|null
+     */
+    protected function getArrayDump(array $arguments, int $number = 0): ?string
+    {
+        if ($number >= 3 || true === empty($arguments)) {
+            return null;
+        }
+
+        if (count($arguments) >= 10) {
+            return (string) count($arguments);
+        }
+
+        $dump = [];
+        foreach ($arguments as $index => $argument) {
+            if ('' === $argument) {
+                $varDump = '(empty string)';
+            } elseif (true === is_scalar($argument)) {
+                $varDump = $this->escapeString($argument);
+            } elseif (true === is_array($argument)) {
+                $varDump = 'Array(' . $this->getArrayDump($argument, $number + 1) . ')';
+            } elseif (true === is_object($argument)) {
+                $varDump = 'Object(' . get_class($argument) . ')';
+            } elseif (null === $argument) {
+                $varDump = 'null';
+            } else {
+                $varDump = $argument;
+            }
+
+            $dump[] = '[' . $index . '] =&gt; ' . $varDump;
+        }
+
+        return implode(', ', $dump);
+    }
+
+    /**
+     * Produces a string representation of a variable
+     *
+     * @param mixed $variable
+     *
+     * @return string
+     */
+    protected function getVarDump($variable): string
```

Comment:
Seems that something is missing in this method. Especially if `$variable` is not `bool` type.